job-events.proto

```diff
@@ -7,15 +7,31 @@ option java_package = "org.cyverse.events.jobs";
 option java_outer_classname = "JobEvents";
 
 message JobEvent {
-    int32 event_id      = 1;
-    string event_name   = 2;
-    string service_name = 3;
-    string host         = 4; // hostname or ip address
-    string app_id       = 5;
-    string job_id       = 6; // id assigned by the DE
-    string job_state    = 7; // only useful for events related to job status updates.
-    string executor_id  = 8; // id assigned by the execution platform (i.e. HTCondor)
-    string user         = 9; // preferably either email address or username.
-    int64 timestamp     = 10; // should be a milliseconds since the epoch
-    string message      = 11;
+    int32 event_id           = 1;
+    string event_name        = 2;
+    string service_name      = 3;
+    string host              = 4; // hostname or ip address
+    string app_id            = 5;
+    required string job_id   = 6; // id assigned by the DE
+    string job_state         = 7; // only useful for events related to job status updates.
+    string executor_id       = 8; // id assigned by the execution platform (i.e. HTCondor)
+    string user              = 9; // preferably either email address or username.
+    int64 timestamp          = 10; // should be a milliseconds since the epoch
+    string message           = 11;
 }
+
+message StepEvent {
+    int32 event_id           = 1;
+    string event_name        = 2;
+    string service_name      = 3;
+    string host              = 4; // hostname or ip address
+    string app_id            = 5;
+    required string job_id   = 6; // id assigned by the DE
+    string job_state         = 7; // only useful for events related to job status updates.
+    string executor_id       = 8; // id assigned by the execution platform (i.e. HTCondor)
+    string user              = 9; // preferably either email address or username.
```

Comment:
I'm not entirely sure how this fits into the conversation, but I think we should put the username/user-id into the message regardless of whether it's placed in the headers. Designing the messages to be as transport-neutral as possible should allow us to switch some or all of the backend services to different transports (like Kafka or Redis) without having change the message formats at the same time.